src/software/sensor_fusion/filter/ball_filter.cpp

```diff
@@ -215,9 +189,46 @@ std::optional<size_t> BallFilter::getAdjustedBufferSize(
     return static_cast<size_t>(buffer_size);
 }
 
-LinearRegressionResults BallFilter::getLinearRegressionLine(
+Line BallFilter::calculateLineOfBestFit(
+    boost::circular_buffer<BallDetection> ball_detections)
+{
+    if (ball_detections.size() < 2)
+    {
+        throw std::invalid_argument("At least 2 elements required for linear regression");
+    }
+
+    auto x_vs_y_regression = calculateLinearRegression(ball_detections);
+
+    // Linear regression cannot fit a vertical line. To get around this, we fit two lines,
+    // one with x and y swapped, so any vertical line becomes horizontal. Then we take the
+    // line of the two that fit the best.
+    boost::circular_buffer<BallDetection> swapped_ball_detections = ball_detections;
+    for (auto &detection : swapped_ball_detections)
+    {
+        detection.position = Point(detection.position.y(), detection.position.x());
+    }
+    auto y_vs_x_regression = calculateLinearRegression(swapped_ball_detections);
+    // Because we swapped the coordinates of the input, we have to swap the coordinates of
+    // the output to get back to our expected coordinate space
+    y_vs_x_regression.regression_line.swapXY();
+
+    // We use the regression from above with the least error
+    Line regression_line =
+        x_vs_y_regression.regression_error < y_vs_x_regression.regression_error
```

Comment:
I think the code conventions suggest avoiding ternary operators

Reply:
fixed